python/lsst/validate/base/datum.py

```diff
@@ -1,5 +1,7 @@
 # See COPYRIGHT file at the top of the source tree.
 from __future__ import print_function, division
+from builtins import str, object
```

Comment:
Probably do not need to import `str`. I see lots of repetitive asserts in the code below here. I think I'd prefer those to all be in a helper function to reduce the amount of code using `basestring`. "is none, bool, int or string" is used a lot.